src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.IServiceProvider.cs

```diff
@@ -10,8 +12,25 @@ internal partial class CSharpProjectShim : Microsoft.VisualStudio.OLE.Interop.IS
     {
         public int QueryService(ref Guid guidService, ref Guid riid, out IntPtr ppvObject)
         {
-            var serviceProvider = (Microsoft.VisualStudio.OLE.Interop.IServiceProvider)ServiceProvider;
-            return serviceProvider.QueryService(ref guidService, ref riid, out ppvObject);
+            if (riid == typeof(IVsContainedLanguageFactory).GUID)
+            {
+                int hr = Shell.ServiceProvider.GlobalProvider.QueryService(guidService, out var serviceObject);
```

Comment:
❓ Why are we using the global provider here instead of the `ServiceProvider` of the current instance?